Src/AutoFakeItEasy/FakeItEasyRelay.cs

```diff
@@ -82,13 +82,28 @@ public object Create(object request, ISpecimenContext context)
 
             var fakeType = typeof(Fake<>).MakeGenericType(type);
 
-            var fake = context.Resolve(fakeType) as FakeItEasy.Configuration.IHideObjectMembers;
+            var fake = context.Resolve(fakeType);
             if (fake == null)
 #pragma warning disable 618
                 return new NoSpecimen(request);
 #pragma warning restore 618
 
-            return fake.GetType().GetProperty("FakedObject").GetValue(fake, null);
+            var resolvedType = fake.GetType();
+            if (!resolvedType.IsFake())
+            {
+#pragma warning disable 618
+                return new NoSpecimen(request);
+#pragma warning restore 618
+            }
+
+            if (resolvedType.GetFakedType() != type)
+            {
+#pragma warning disable 618
+                return new NoSpecimen(request);
+#pragma warning restore 618
+            }
+
+            return resolvedType.GetProperty("FakedObject").GetValue(fake, null);

```

Comment:
Left to my own devices, I would've combined the three previous ifs into one, (or at least the last two), but that doesn't seem to be the way the other code was written, so I kept it separate.

I suppose another option would be to see if `resolvedType.GetFakedType()` was assignable to `type`, as I guess returning a `Fake<Derived>` when asked for a `Fake<Base>` could be a valid use case, depending on how AutoFixure's been configured.